app/scripts/metamask-controller.js

```diff
@@ -522,6 +522,8 @@ export default class MetamaskController extends EventEmitter {
         ApprovalType.WatchAsset,
         ApprovalType.EthGetEncryptionPublicKey,
         ApprovalType.EthDecrypt,
+        // Exclude Smart TX Status Page from rate limiting to allow sequential transactions
+        'smartTransaction:showSmartTransactionStatusPage',
```

Comment:
Nice job with the fix and detailed description.

nit: can you move this into ApprovalType, so it's consistent with the rest of the list?

Reply:
Well, I thought about this, but quickly realized that, this isn't actually an `ApprovalType` - it's a custom string identifier for the Smart TX status page. The `typesExcludedFromRateLimiting` array accepts both `ApprovalType` enum values and "custom string identifiers".

It's just a UI notification we want to exclude from rate limiting.

Keeping it as a string with a clear comment I think is all we need.

Reply:
I should have searched for the string. Good catch!